File: Python/level1/소수 찾기.py
def solution(n):
    numbers = [1,1] + [0] * (n - 1)
    for index in range(2, n+1):
        if index == 0 or index == 1:
            continue
        if numbers[index] == 0:
            for number in range(index*2,n+1,index):
                numbers[number] = 1
    return len([index for number in numbers if number == 0])

# def solution(n):
#     num=set(range(2,n+1))

#     for i in range(2,n+1):
#         if i in num:
#             num-=set(range(2*i,n+1,i))
#     return len(num)
# set은 차집합이 가능하다

File: Cplusplus/LEVEL 2/소수 찾기.cpp
//22.02.08 소수 찾기 by CJHofficial

#include <string>
#include <vector>
#include <algorithm>

using namespace std;
bool checkPrime(int num) {
    if (num == 0 || num == 1) return false;
    for (int i = 2; i < num; i++) {
        if (num % i == 0) return false;
    }
    return true;
}
int solution(string numbers) {
    int answer = 0;
    vector<int> answerList;
    sort(numbers.begin(), numbers.end());
    for (int i = 1; i <= numbers.size(); i++) {
        vector<bool> picked(numbers.size(), false);
        for (int j = 0; j < i; j++) picked[j] = true;
        do {
            string temp = "";
            for (int j = 0; j < numbers.size(); j++) {
                if (picked[j]) temp += numbers[j];
            }
            int num = stoi(temp);
            if (checkPrime(num)) answerList.push_back(num);
        } while (next_permutation(numbers.begin(), numbers.end()));
    }
    sort(answerList.begin(), answerList.end());
    answerList.erase(unique(answerList.begin(), answerList.end()), answerList.end());
    answer = answerList.size();
    return answer;
}